lib/omniauth/strategies/slack.rb

```diff
@@ -19,56 +19,37 @@ class Slack < OmniAuth::Strategies::OAuth2
         param_name: 'token'
       }
 
-      uid { raw_info['user_id'] }
+      # User ID is not guaranteed to be globally unique across all Slack users.
+      # The combination of user ID and team ID, on the other hand, is guaranteed
+      # to be globally unique.
+      uid { "#{user_identity['id']}-#{team_identity['id']}" }
 
       info do
         hash = {
-          nickname: raw_info['user'],
-          team: raw_info['team'],
-          user: raw_info['user'],
-          team_id: raw_info['team_id'],
-          user_id: raw_info['user_id']
```

Comment:
I know its not guaranteed to be unique across all Slack, but you very likely to need the user_id if you want to use RTM API based on the data you get here. So, imo we should keep `user_id` and `team_id`.